react/src/components/PostHogFeature.tsx

```diff
@@ -0,0 +1,94 @@
+import { useFeatureFlagPayload, useFeatureFlagVariantKey, usePostHog } from '../hooks'
+import React, { RefObject, useEffect, useMemo, useRef, useState } from 'react'
+import { PostHog } from '../context'
+
+export type PostHogFeatureProps = {
+    flag: string
+    children: React.ReactNode | ((payload: any) => React.ReactNode)
+    match?: string | boolean
+    visibilityObserverOptions?: IntersectionObserverInit
+}
+
+export function PostHogFeature({
```

Comment:
It would be lovely if we could include a code example somewhere within the library on how this is intended to be used 🥺

Reply:
Will go in the docs yes!

Reply:
https://github.com/PostHog/posthog.com/pull/6043